### ISSUE TITLE
More configuration options for routes in projects.

### ISSUE THREAD

**Issue by SyneticDevOps** [uselagoon/lagoon#2956]
TLDR;
- Disable routes on a per service (```routes.services.keycloak.enabled: false``` or something)
- Disable auto route generation per service (```routes.services.keycloak.auto_generation: false``` or something)
- Get routes per service from the API  (something like ```{services: { nginx: { routes: 'https://example.org' }, 'keycloak': { routes: 'https://keycloak.example.org' } } }```)

We have various situations where we want to add services, such as a keycloak service, gotenberg pdf service, mailhog, etc. Those have an public route available to them. But we do not want the auto route generation to create dynamic routes for them as it does for the Drupal nginx service. It can create quite a hassle with all the SSL certificates it needs to get from letsencrypt, and integration into our CI/CD is more complicated as well. In our CI/CD we try to get all the routes available, and run accessiblity and lighthouse tests on them. 

For instance, we have a project with 40 routes (Drupal with Domain Access) and also a gotenberg service. Currently on the dev environments we have 80 routes generated (instead of 40), through the prefixes for the nginx and gotenberg service.
In this case we want the gotenberg service to have no public routes at all.

Or we want to run mailhog on the dev environments, and access it on mailhog.(the dev route), and prevent various prefixed routes that give strange results when running our CI/CD tools.

**Comment by jbertoen** [comment]
+1

**Comment by tobybellwood** [comment]
Does this work for you?
https://docs.lagoon.sh/lagoon/using-lagoon-the-basics/lagoon-yml#routes.autogenerate.enabled
```
routes:
  autogenerate:
    enabled: false
```

On creating services for the different (non-standard) Lagoon apps - we're looking at ways to better support flexibility in this, the current templating is pretty restrictive around exposed ports etc

**Comment by jbertoen** [comment]
This disables autogeneration on all services right? I want it enabled on nginx, but disabled on other route enabled services.

For instance I have 2 nginx containers (for some complicated high tech fancy pancy reason). One is my drupal nginx container, with Domain Access module setup. Which could have like 1-X number of routes (which already works, awesome). The other nginx container is perhaps an internal container (only accessible from within the local network on k8s) or it only has one fixed route per environment. Both these cases are not possible now.

We currently use the lagoon.type: node to integrate this on amazee: https://github.com/gotenberg/gotenberg.
This service exposes an API to generate pdf's (like AthenaPDF, but better), but in our case we do not want a public route available to access the API. We only want our services within the project (like a PHP container) to call it through the hostname. Or perhaps (for some reason) we do want it to be exposed, but on a fixed route. But the Nginx route has prefixed autogenerated routes.

**Comment by tobybellwood** [comment]
Hmm, ok - I'll have to check the code (we've *just* rewritten it!) to see if selective re-enable is possible, or if there's a better way!

**Comment by jbertoen** [comment]
That would be awesome, it would give more fine grained control about route generation en also minimize the number of letsencrypt certificates that need to be generated on dynamic environments.

**Comment by tobybellwood** [comment]
The docs for per-service route disabling are at https://docs.lagoon.sh/using-lagoon-the-basics/docker-compose-yml/#auto-generated-routes

Currently Lagoon has no specific service knowledge to enable per-service disabling via the api, but we can consider this as part of our path to first-class routes (https://github.com/uselagoon/lagoon/discussions/3467)
